### PR TITLE
Move ActiveRecord serialize coder finder to own method.

### PR DESCRIPTION
**Pull request by simi**
I got this idea while I was finding something in source code. I hope it makes sense.

### CODE REVIEW

File: activerecord/lib/active_record/attribute_methods/serialization.rb [hunk 1]
```diff
@@ -34,21 +34,27 @@ module ClassMethods
...
+          coder = serialize_coder_for(class_name_or_coder)
 
           decorate_attribute_type(attr_name, :serialize) do |type|
             Type::Serialized.new(type, coder)
           end
         end
+
+        private
+
+        def serialize_coder_for(class_name_or_coder)
```
Comment (mapreal19) [inline activerecord/lib/active_record/attribute_methods/serialization.rb]: Methods under private/protected should be indented :)

File: activerecord/lib/active_record/attribute_methods/serialization.rb [hunk 2]
```diff
@@ -34,21 +34,27 @@ module ClassMethods
...
+          # When ::JSON is used, force it to go through the Active Support JSON encoder
+          # to ensure special objects (e.g. Active Record models) are dumped correctly
+          # using the #as_json hook.
+          if class_name_or_coder == ::JSON
```
Comment (mapreal19) [inline activerecord/lib/active_record/attribute_methods/serialization.rb]: Just an idea, I found easier to read here if we use guard clauses:

    return Coders::JSON if class_name_or_coder == ::JSON
    return class_name_or_coder if [:load, :dump].all? { |x| class_name_or_coder.respond_to?(x) }
    
    Coders::YAMLColumn.new(class_name_or_coder)

File: activerecord/lib/active_record/attribute_methods/serialization.rb (hunk 1 above)
Comment (simi) [inline activerecord/lib/active_record/attribute_methods/serialization.rb]: Ahh, good point!

**Comment by sgrif** [comment]
Thanks, but this qualifies as a cosmetic change, which we don't accept.

**Comment by simi** [comment]
@sgrif actually it is not cosmetic change for me since I needed this separately and I needed to copy this part to my code. Maybe making it public makes sense here.

**Comment by sgrif** [comment]
This is definitely not something I'd like to add to the public API.

**Comment by simi** [comment]
OK, thanks for review.

**Pull request closed without merging**